### DISCUSSION TITLE
2023-10-23 개발 방향 회의

### DISCUSSION THREAD

**Discussion by chanhihi** [jiphyeonjeon-42/frontend#579] [meeting]
개발 방향 회의 입니다.

- https://github.com/orgs/jiphyeonjeon-42/discussions/31

**Comment by chanhihi** [comment]
파일 구조 변경

**Comment by not-using** [comment]
마감기한

**Reply by chanhihi** [reply]
필요하다

**Comment by not-using** [comment]
기술스택, 특히 css 방식 논의

**Reply by not-using** [reply]
### 현재 상황의 문제점
- css 파일이 떨어져 있어서 보기 힘들다
- 다른 컴포넌트에 영향을 주는 경우가 있었음 모듈화 필요
- eslint랑 prettier 적용이 제대로 되지 않음
- test 부재
- type 이상함

**Reply by mink97** [reply]
스토리북?

**Reply by not-using** [reply]
기존 프로젝트가 복잡해서 이해하는 시간이 오래걸림 
-> 새로 만들어보면 어떨까 프레임워크를 도입하면 폴더구조 등의 설정 도움을 받을 수 있음 nextjs
-> 처음부터 전부 시도하는것보다 프레임워크 이해부터 필요함
-> 온보딩 형식으로 스터디를 해보자

**Reply by not-using** [reply]
## CSS 방식 수정 => 기각 
https://nextjs.org/docs/app/building-your-application/styling

**Comment by not-using** [comment]
온보딩 준비하기 
참고: https://github.com/42organization

**Reply by mink97** [reply]
프론트 지원하는 사람이 없다면?

**Reply by chanhihi** [reply]
온보딩 준비하면서 다같이 처음부터 시작하는 느낌으로 진행해보기

**Reply by not-using** [reply]
## 우리끼리 시도해보는 Next.js 
### 목적
- nextjs 공부
- 현재 페이지를 next로 바꾸는 작업이 할만한지 파악

### 구현 페이지
현재 42library.kr 클론코딩

- 메인페이지
- 도서 검색페이지
- 도서 상세페이지 
- (선택) 로그인 및 헤더
- (선택) 반응형

### 구현시 참고
- CSR, SPA 방식은 유지함/  SSG, SSR 무시 (https://nextjs.org/docs/app/building-your-application/deploying/static-exports)
